google/cloud/spanner/value.h

```diff
@@ -363,7 +365,27 @@ class Value {
   static bool TypeProtoIs(std::string const&, google::spanner::v1::Type const&);
   static bool TypeProtoIs(Bytes const&, google::spanner::v1::Type const&);
   static bool TypeProtoIs(Json const&, google::spanner::v1::Type const&);
-  static bool TypeProtoIs(Numeric const&, google::spanner::v1::Type const&);
+  template <DecimalMode Mode>
+  static bool TypeProtoIs(Decimal<Mode> const&,
+                          google::spanner::v1::Type const& type) {
```

Comment:
@devjgm ... You seemed intent on making these templates, with conditional behavior (albeit at compile time), so I've done that.  For my money they are clearer and cleaner as non-template overloads for the two candidates, `Numeric` and `PgNumeric`.  But I'll let you decide.

Reply:
I don't recall exactly the discussion about this function. I do recall having a preference for `Decimal<M>` being a template rather than two separate, non-templated classes, which is indeed what you did, so I'm happy.

At this point, I have no preference about whether this function is a template or overloaded. Either is fine with me. Happy if you change it or keep it, I'll approve either way.

Reply:
OK, thanks.  Switched to overloads.